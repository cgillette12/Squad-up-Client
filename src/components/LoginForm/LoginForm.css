.login-parent{
   margin-top: 3vh;
     align-items: center;
     justify-content: center;
     display: flex;
     flex-direction: column;
     line-height: 1.8;
   
}
.login-section{
  width:40%;
  line-height: 2;
  display: flex;
  align-items: center;
  flex-direction: column;
  align-content: center;
  justify-content: space-evenly;
    
}
.login-form{
  width: 452px;
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
  height: 350px;

} 
.login-button{
  font-size:18px;
  height: 42px;
}
#login-username-input, #login-password-input{
  font-size:18px;
  height:42px;
  width:100%
}

  