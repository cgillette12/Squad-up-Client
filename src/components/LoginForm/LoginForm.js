import React, { useState, useContext } from 'react'
import AuthApiService from '../../services/auth-api-service'
import UserContext from '../../contexts/UserContext'
import { Input } from '../../components/FormUtils/FormUtils'
import './LoginForm.css'
import DemoCreds from '../DemoCreds/DemoCreds';

export default function LoginForm(props) {
  const [user, setUser] = useState('')
  const [password, setPassword] = useState('')
  const [error, setError] = useState(null)
  const { onLoginSuccess = () => { } } = props;
  const context = useContext(UserContext);

  const handleSubmit = e => {
    e.preventDefault()

    AuthApiService.postLogin({
      username: user,
      password: password,
    })
      .then(res => {
        setUser('')
        setPassword('')
        context.processLogin(res.authToken)
        onLoginSuccess()
      })
      .catch(res => {
        setError(res.error)
      })
  }

  return (
        <form
          className="login-form"
          onSubmit={handleSubmit}
        >
          <div className="red" role='alert'>
            {error && <p>{error}</p>}
          </div>
            <Input
              id="login-username-input"
              type="text"
              name="username"
              placeholder='Username'
              value={user}
              onChange={e => setUser(e.target.value)}
              aria-label='username'
              autoFocus
              required
            />
            <Input
              id="login-password-input"
              type="password"
              name="password"
              placeholder='Password'
              value={password}
              onChange={e => setPassword(e.target.value)}
              aria-label='password'
              required
            />
          <button type="submit" className="login-button">
            Login
          </button>
          <DemoCreds />
    </form>
  )
}
