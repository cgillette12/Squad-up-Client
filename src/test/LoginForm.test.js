import React from 'react';
import {BrowserRouter} from 'react-router-dom'
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import LoginForm from '../components/LoginForm/LoginForm';



it('renders LoginForm without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
        <BrowserRouter>
        <LoginForm />
        </BrowserRouter>, div)
    ReactDOM.unmountComponentAtNode(div)
})

it('renders the LandingPage as expected', () => {
    const tree = renderer
        .create(<BrowserRouter>
            <LoginForm/>
        </BrowserRouter>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});


