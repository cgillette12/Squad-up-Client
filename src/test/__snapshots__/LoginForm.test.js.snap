// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the LandingPage as expected 1`] = `
<form
  className="login-form"
  onSubmit={[Function]}
>
  <div
    className="red"
    role="alert"
  />
  <input
    aria-label="username"
    autoFocus={true}
    id="login-username-input"
    name="username"
    onChange={[Function]}
    placeholder="Username"
    required={true}
    type="text"
    value=""
  />
  <input
    aria-label="password"
    id="login-password-input"
    name="password"
    onChange={[Function]}
    placeholder="Password"
    required={true}
    type="password"
    value=""
  />
  <button
    className="login-button"
    type="submit"
  >
    Login
  </button>
  <button
    className="link-button"
    onClick={[Function]}
  >
    Demo
  </button>
</form>
`;
